packages/fxa-auth-server/lib/verification-reminders.js

```diff
@@ -165,23 +165,28 @@ module.exports = (log, config) => {
             const cutoff = now - intervals[key];
             result[key] = redis
               .zpoprangebyscore(key, 0, cutoff, 'WITHSCORES')
-              .reduce(async (reminders, item, index) => {
-                if (index % 2 === 0) {
-                  const uid = item;
-                  let metadata = await redis.get(`${METADATA_KEY}:${uid}`);
-                  if (metadata) {
-                    const [flowId, flowBeginTime] = JSON.parse(metadata);
-                    metadata = { flowId, flowBeginTime };
-                    if (keyIndex === keys.length - 1) {
-                      await redis.del(`${METADATA_KEY}:${uid}`);
+              .then(async items => {
```

Comment:
switching to native promises means a slight refactor; i used a `for` loop

Reply:
`items.reduce` should work right?  I don't understand why a `for...of` loop is necessary.

Reply:
Array.reduce isn't async afaik. I seem to remember trying that

Reply:
Ah, ok.  Got it.  👍